src/domains/TypesDomain.js

```diff
@@ -50,8 +51,9 @@ export default class TypesDomain {
         if (lType === undefined || rType === undefined) return TypesDomain.topVal;
         if (Value.isTypeCompatibleWith(lType, StringValue) || Value.isTypeCompatibleWith(rType, StringValue))
           resultType = StringValue;
-        else if (Value.isTypeCompatibleWith(lType, NumberValue) && Value.isTypeCompatibleWith(rType, NumberValue))
-          resultType = NumberValue;
+        else if (lType === IntegralValue && rType === IntegralValue) resultType = IntegralValue;
+        else if (Value.isTypeCompatibleWith(lType, NumberValue) && Value.isTypeCompatibleWith(rType, NumberValue));
```

Comment:
The semicolon at the end of this line should not be there.

Reply:
oh my fault :XD